developer/src/kmc-kmn/src/compiler/kmn-compiler-messages.ts

```diff
@@ -182,226 +213,525 @@ export class CompilerMessages {
   static WARN_HeaderStatementIsDeprecated                     = SevWarn | 0x09D;
+
+  static Warn_UseNotLastStatementInRule                       = () => m(this.WARN_UseNotLastStatementInRule, `A rule with use() statements in the output should not have other content following the use() statements`);
   static WARN_UseNotLastStatementInRule                       = SevWarn | 0x09E;
 
+  static Warn_TouchLayoutFontShouldBeSameForAllPlatforms = () => mw(this.WARN_TouchLayoutFontShouldBeSameForAllPlatforms,
+    `The touch layout font should be the same for all platforms.`);
   static WARN_TouchLayoutFontShouldBeSameForAllPlatforms      = SevWarn | 0x09F;
-  static WARN_InvalidJSONMetadataFile                         = SevWarn | 0x0A0;
-  static WARN_JSONMetadataOSKFontShouldMatchTouchFont         = SevWarn | 0x0A1;
+
+  // static WARN_InvalidJSONMetadataFile                         = SevWarn | 0x0A0;
+  // static WARN_JSONMetadataOSKFontShouldMatchTouchFont         = SevWarn | 0x0A1;
+
+  static Warn_KVKFileIsInSourceFormat                         = () => m(this.WARN_KVKFileIsInSourceFormat, `.kvk file should be binary but is an XML file`);
   static WARN_KVKFileIsInSourceFormat                         = SevWarn | 0x0A2;
 
+  // kmcmplib: static Warn_DontMixChiralAndNonChiralModifiers              = () => m(this.WARN_DontMixChiralAndNonChiralModifiers, `Don't mix the use of left/right modifiers with non-left/right modifiers in the same platform`);
+  static Warn_DontMixChiralAndNonChiralModifiers = () => mw(this.WARN_DontMixChiralAndNonChiralModifiers,
+    `This keyboard contains Ctrl,Alt and LCtrl,LAlt,RCtrl,RAlt sets of modifiers. Use only one or the other set for web target.`);
   static WARN_DontMixChiralAndNonChiralModifiers              = SevWarn | 0x0A3;
+
+  static Warn_MixingLeftAndRightModifiers                     = () => m(this.WARN_MixingLeftAndRightModifiers, `Left and right modifiers should not both be used in the same rule`);
   static WARN_MixingLeftAndRightModifiers                     = SevWarn | 0x0A4;
 
+  static Warn_LanguageHeadersDeprecatedInKeyman10             = () => m(this.WARN_LanguageHeadersDeprecatedInKeyman10, `This language header has been deprecated in Keyman 10. Instead, add language metadata in the package file`);
   static WARN_LanguageHeadersDeprecatedInKeyman10             = SevWarn | 0x0A5;
 
+  static Hint_NonUnicodeFile                                  = () => m(this.HINT_NonUnicodeFile, `Keyman Developer has detected that the file has ANSI encoding. Consider converting this file to UTF-8`);
   static HINT_NonUnicodeFile                                  = SevHint | 0x0A6;
 
-  static WARN_TooManyErrorsOrWarnings                         = SevWarn | 0x0A7;
+  // static WARN_TooManyErrorsOrWarnings                         = SevWarn | 0x0A7;
```

Comment:
I notice some of the WARNs must no longer be used?

Reply:
maybe left commented out to preserve the number? 
elsewhere i had just put a comment, "//there's a gap at this range.."

Reply:
Yes, commented out because they are unused and in 18.0 we may come back and consolidate. I didn't want to diverge into deeper analysis of message usage.